File: packages/dcflight/ios/Classes/Theme/DCFTheme.swift
/*
 * Copyright (c) Dotcorr Studio. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import UIKit

/// Unified theme system for DCFlight iOS
/// Provides single API for all theme operations (matches Android DCFTheme)
public class DCFTheme {
    
    /// Check if current theme is dark mode
    public static func isDarkTheme(traitCollection: UITraitCollection = UITraitCollection.current) -> Bool {
        return traitCollection.userInterfaceStyle == .dark
    }
    
    /// Get system text color (adapts to light/dark mode)
    public static func getTextColor(traitCollection: UITraitCollection = UITraitCollection.current) -> UIColor {
        return UIColor.label
    }
    
    /// Get system background color (adapts to light/dark mode)
    public static func getBackgroundColor(traitCollection: UITraitCollection = UITraitCollection.current) -> UIColor {
        return UIColor.systemBackground
    }
    
    /// Get system surface color (adapts to light/dark mode)
    public static func getSurfaceColor(traitCollection: UITraitCollection = UITraitCollection.current) -> UIColor {
        return UIColor.secondarySystemBackground
    }
    
    /// Get system accent color
    public static func getAccentColor(traitCollection: UITraitCollection = UITraitCollection.current) -> UIColor {
        return UIColor.systemBlue
    }
    
    /// Get secondary text color (adapts to light/dark mode)
    public static func getSecondaryTextColor(traitCollection: UITraitCollection = UITraitCollection.current) -> UIColor {
        return UIColor.secondaryLabel
    }
    
    /// Get all theme colors as a dictionary (for Dart bridge)
    public static func getAllColors(traitCollection: UITraitCollection = UITraitCollection.current) -> [String: Any] {
        let isDark = isDarkTheme(traitCollection: traitCollection)
        let textColor = getTextColor(traitCollection: traitCollection)
        let backgroundColor = getBackgroundColor(traitCollection: traitCollection)
        let surfaceColor = getSurfaceColor(traitCollection: traitCollection)
        let accentColor = getAccentColor(traitCollection: traitCollection)
        let secondaryTextColor = getSecondaryTextColor(traitCollection: traitCollection)
        
        return [
            "isDark": isDark ? 1 : 0,
            "textColor": colorToHex(textColor),
            "backgroundColor": colorToHex(backgroundColor),
            "surfaceColor": colorToHex(surfaceColor),
            "accentColor": colorToHex(accentColor),
            "secondaryTextColor": colorToHex(secondaryTextColor)
        ]
    }
    
    /// Convert UIColor to hex string
    private static func colorToHex(_ color: UIColor) -> String {
        var red: CGFloat = 0
        var green: CGFloat = 0
        var blue: CGFloat = 0
        var alpha: CGFloat = 0
        
        color.getRed(&red, green: &green, blue: &blue, alpha: &alpha)
        
        let r = Int(red * 255)
        let g = Int(green * 255)
        let b = Int(blue * 255)
        let a = Int(alpha * 255)
        
        return String(format: "#%02X%02X%02X%02X", a, r, g, b)
    }
}

File: packages/template/examples/reconciliation_test/ios/.symlinks/plugins/dcflight/android/src/main/kotlin/com/dotcorr/dcflight/theme/DCFTheme.kt
/*
 * Copyright (c) Dotcorr Studio. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.dotcorr.dcflight.theme

import android.content.Context
import android.content.res.Configuration
import android.graphics.Color
import android.os.Build
import android.util.TypedValue
import com.dotcorr.dcflight.utils.AdaptiveColorHelper

/**
 * Unified theme system for DCFlight Android
 * Provides single API for all theme operations (replaces scattered theme code)
 */
object DCFTheme {
    
    /**
     * Check if current theme is dark mode
     */
    fun isDarkTheme(context: Context): Boolean {
        return AdaptiveColorHelper.isDarkTheme(context)
    }
    
    /**
     * Get system text color (adapts to light/dark mode)
     */
    fun getTextColor(context: Context): Int {
        return AdaptiveColorHelper.getSystemTextColor(context)
    }
    
    /**
     * Get system background color (adapts to light/dark mode)
     */
    fun getBackgroundColor(context: Context): Int {
        return AdaptiveColorHelper.getSystemBackgroundColor(context)
    }
    
    /**
     * Get system surface color (adapts to light/dark mode)
     */
    fun getSurfaceColor(context: Context): Int {
        return try {
            val typedValue = TypedValue()
            if (context.theme.resolveAttribute(android.R.attr.colorBackground, typedValue, true)) {
                typedValue.data
            } else {
                if (isDarkTheme(context)) {
                    Color.parseColor("#121212") // Material dark surface
                } else {
                    Color.WHITE
                }
            }
        } catch (e: Exception) {
            if (isDarkTheme(context)) {
                Color.parseColor("#121212")
            } else {
                Color.WHITE
            }
        }
    }
    
    /**
     * Get system accent color
     */
    fun getAccentColor(context: Context): Int {
        return AdaptiveColorHelper.getSystemAccentColor(context)
    }
    
    /**
     * Get secondary text color (adapts to light/dark mode)
     */
    fun getSecondaryTextColor(context: Context): Int {
        return try {
            val typedValue = TypedValue()
            if (context.theme.resolveAttribute(android.R.attr.textColorSecondary, typedValue, true)) {
                typedValue.data
            } else {
                if (isDarkTheme(context)) {
                    Color.parseColor("#B3FFFFFF") // 70% opacity white
                } else {
                    Color.parseColor("#8A000000") // 54% opacity black
                }
            }
        } catch (e: Exception) {
            if (isDarkTheme(context)) {
                Color.parseColor("#B3FFFFFF")
            } else {
                Color.parseColor("#8A000000")
            }
        }
    }
    
    /**
     * Get all theme colors as a map (for Dart bridge)
     */
    fun getAllColors(context: Context): Map<String, Int> {
        return mapOf(
            "isDark" to if (isDarkTheme(context)) 1 else 0,
            "textColor" to getTextColor(context),
            "backgroundColor" to getBackgroundColor(context),
            "surfaceColor" to getSurfaceColor(context),
            "accentColor" to getAccentColor(context),
            "secondaryTextColor" to getSecondaryTextColor(context)
        )
    }
}

